src/types.rs

```diff
@@ -175,6 +175,33 @@ pub trait Interface: Send + Sync + InterfaceClone {
     fn generate_event(&self, _event: String) -> Result<()> {
         unimplemented!("generate_event")
     }
+
+    /// Sends an async message
+    ///
+    /// # Arguments
+    ///
+    /// * `target_address` - Destination address hash in format string
+    /// * `target_handler` - Name of the message handling function
+    /// * `validity_start` - Tuple containing the period and thread of the validity start slot
+    /// * `validity_end` - Tuple containing the period and thread of the validity end slot
+    /// * `max_gas` - Maximum gas for the message execution
+    /// * `gas_price` - Price of one gas unit
+    /// * `coins` - Coins of the sender
+    /// * `payload` - Message payload
+    ///
+    fn send_message(
+        &self,
+        target_address: &str,
+        target_handler: &str,
+        validity_start: (u64, u64),
+        validity_end: (u64, u64),
+        max_gas: u64,
+        gas_price: &str,
+        coins: &str,
+        payload: &str,
```

Comment:
rename "payload" to "data" and let it be a `&[u8]` like we do for set_data for example